### PR TITLE
Swapped out String type for URL type in config structs

### PR DESCRIPTION
**Pull request by matthiasdebernardini**
Was working directly off of master and this makes it weird to rebase so thankfully @okjodom helped me out by showing me how to rebase.

Also many thanks to @NicolaLS for his intuition during debugging.

This PR is for issues #376 previous PR is #466 - sorry for the confusion.

### CODE REVIEW

File: client/client-lib/src/api.rs [hunk 1]
```diff
@@ -253,6 +254,8 @@ impl<C: JsonRpcClient + Send + Sync> FederationApi for WsFederationApi<C> {
     }
 
     async fn fetch_gateways(&self) -> Result<Vec<LightningGateway>> {
+        // println!("{:?}", self.request("/ln/list_gateways", ()).await);
```
Comment (okjodom) [inline client/client-lib/src/api.rs:257]: nit: remove commented debug line?

**Review by okjodom — APPROVED** [review]
Looks good!

File: client/client-lib/src/api.rs (hunk 1 above)
Comment (matthiasdebernardini) [reply client/client-lib/src/api.rs]: done

**Review by justinmoon — COMMENTED** [review]
Looks good, a couple nits. Fix these and we'll merge.

File: scripts/latency-test.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@
 set -eu
 FM_FED_SIZE=${1:-4}
 ITERATIONS=${2:-5}
-export RUST_LOG=error,ln_gateway=off
+export RUST_LOG=debug,ln_gateway=off
```
Comment (NicolaLS) [inline scripts/latency-test.sh:7]: is this on purpose ?

File: client/client-lib/src/lib.rs [hunk 3]
```diff
@@ -629,7 +629,13 @@ impl Client<UserClientConfig> {
     pub async fn await_outgoing_contract_execution(&self, contract_id: ContractId) -> Result<()> {
         let gateway = self.fetch_active_gateway().await?;
         let future = reqwest::Client::new()
-            .post(&format!("{}pay_invoice", gateway.api))
+            .post(
+                gateway
+                    .api
+                    .join("pay_invoice")
+                    .expect("Could not join path to Url")
```
Comment (justinmoon) [inline client/client-lib/src/lib.rs:636]: Please change this to a `?`

**Review by justinmoon — CHANGES_REQUESTED** [review]

File: client/client-lib/src/lib.rs (hunk 3 above)
Comment (matthiasdebernardini) [reply client/client-lib/src/lib.rs]: I assume since its better to use the library error in this case?

**Review by justinmoon — DISMISSED** [review]
LGTM! Nice work @matthiasdebernardini!

**Review by justinmoon — APPROVED** [review]

**Pull request merged**